### ISSUE TITLE
Cannot run singularity: failed to set securebits

### ISSUE THREAD

**Issue by letmaik** [microsoft/WSL#4278]
**Please fill out the below information:**
* Your Windows build number:  (Type `ver` at a Windows Command Prompt)
10.0.18362.207

* What you're doing and what's happening:  (Copy&paste the full set of _specific_ command-line steps necessary to reproduce the behavior, and their output. Include screen shots if that helps demonstrate the problem.)
```sh
$ singularity --version
singularity version 3.2.1-1
$ singularity run library://sylabsed/examples/lolcow
ERROR  : Failed to set securebits: Invalid argument
```

* What's wrong / what should be happening instead:

Singularity is [trying to set securebits](https://github.com/sylabs/singularity/blob/42a714ae14fccf14dd2d3301ec2911493c262638/cmd/starter/c/starter.c#L220-L222) but fails:
```c
    if ( prctl(PR_SET_SECUREBITS, SECBIT_NO_SETUID_FIXUP|SECBIT_NO_SETUID_FIXUP_LOCKED) < 0 ) {
        fatalf("Failed to set securebits: %s\n", strerror(errno));
    }
```

Related: https://github.com/microsoft/WSL/issues/3641

**Comment by Biswa96** [comment]
Are you using WSL2? You can get it from `wsl.exe -l -v` command.

**Comment by letmaik** [comment]
@Biswa96 No I'm not. I'm not running Windows Insiders. Would be good to clarify if it will ever be supported in WSL 1. If not, then this issue can be closed.

**Comment by michoo-keyman** [comment]
Folks, I have been seeing this issue two weeks now. It looks to me that some read/write error of some sort happened to libdvd-pkg.

All I did to get this error is to run on WLS the update commands like I used to do in Ubuntu:
sudo apt update && sudo apt upgrade -y

I then get this error:
etting up libdvd-pkg (1.4.2-1-1) ...
libdvd-pkg: Checking orig.tar integrity...
/usr/src/libdvd-pkg/libdvdcss_1.4.2.orig.tar.bz2: OK
libdvd-pkg: Unpacking and configuring...
libdvd-pkg: Building the package... (it may take a while)
libdvd-pkg: Build log will be saved to /usr/src/libdvd-pkg/libdvdcss2_1.4.2-1~local_amd64.build
failed to set securebits to 024/0x14
dpkg: error processing package libdvd-pkg (--configure):
 installed libdvd-pkg package post-installation script subprocess returned error exit status 1
libdvd-pkg: Checking orig.tar integrity...
/usr/src/libdvd-pkg/libdvdcss_1.4.2.orig.tar.bz2: OK
libdvd-pkg: Unpacking and configuring...
libdvd-pkg: Building the package... (it may take a while)
libdvd-pkg: Build log will be saved to /usr/src/libdvd-pkg/libdvdcss2_1.4.2-1~local_amd64.build
failed to set securebits to 024/0x14
E: Sub-process /usr/bin/dpkg returned an error code (2)

Please tell me what else you need to fix this? 
Thx

**Comment by ChaosFreak** [comment]
I'm still seeing this issue using WSL / Ubuntu 20.04.1 LTS (focal).

```
$ sudo apt-get install libdvd-pkg
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following NEW packages will be installed:
  libdvd-pkg
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/14.9 kB of archives.
After this operation, 80.9 kB of additional disk space will be used.
Preconfiguring packages ...
Selecting previously unselected package libdvd-pkg.
(Reading database ... 99222 files and directories currently installed.)
Preparing to unpack .../libdvd-pkg_1.4.2-1-1_all.deb ...
Unpacking libdvd-pkg (1.4.2-1-1) ...
Setting up libdvd-pkg (1.4.2-1-1) ...
libdvd-pkg: Checking orig.tar integrity...
/usr/src/libdvd-pkg/libdvdcss_1.4.2.orig.tar.bz2: OK
libdvd-pkg: Unpacking and configuring...
libdvd-pkg: Building the package... (it may take a while)
libdvd-pkg: Build log will be saved to /usr/src/libdvd-pkg/libdvdcss2_1.4.2-1~local_amd64.build
failed to set securebits to 024/0x14
dpkg: error processing package libdvd-pkg (--configure):
 installed libdvd-pkg package post-installation script subprocess returned error exit status 1
dpkg: error: failed to remove my own update file /var/lib/dpkg/updates/0000: No such file or directory
E: Sub-process /usr/bin/dpkg returned an error code (2)
```

Do you need any other information?

**Comment by ianferreira** [comment]
Any update on this issue
INFO:    Starting build...
INFO:    Use cached image
WARNING: failed to set O_CLOEXEC flags on image
FATAL:   While performing build: packer failed to pack: root filesystem extraction failed: extract command failed: WARNING: failed to set O_CLOEXEC flags on image
WARNING: failed to set O_CLOEXEC flags on image
ERROR  : Failed to set securebits: Invalid argument

**Comment by ianferreira** [comment]
@therealkenc Hello?

**Comment by therealkenc** [comment]
![image](https://user-images.githubusercontent.com/7049700/115477842-add0c100-a1f9-11eb-8809-0d9f032ef8c8.png)

**Comment by bgroenks96** [comment]
This is still happening for me in version 3.8.1?

```
~/workspace/sw/singularity-3.8.1$ singularity shell lolcow_latest.sif
WARNING: failed to set O_CLOEXEC flags on image
WARNING: failed to set O_CLOEXEC flags on image
ERROR  : Failed to set securebits: Invalid argument
```

I am using WSL2. Do I need to update it somehow?

**Comment by Eddiiie** [comment]
Same here.  How to fix?   Can't install anything or upgrade.  HElp!!

**Comment by williamty** [comment]
I figured this issue by "sudo".

**Comment by LorenzoMerotto** [comment]
Still have this problem and none of the solutions shown are helpful

**Comment by suviu001** [comment]
Suggest u update wsl1 to wsl2，use this command: wsl --set-version Ubtuntu-22.04 2. Singularity running di ok on type wsl2.

**Comment by Allanprince001** [comment]
Hi, I am using windows version 10.0.19045.4170 and WSL
ubuntu version 22.04.3 (jammy)

Following the website https://apptainer.org/docs/admin/main/installation.html#install-from-pre-built-packages, I tried to install apptainer from the source. Following the procedure there, I installed the system dependancies, Installed Go, cloned the repo. While compiling the Apptainer, I was able to execute the commands ./mconfig and cd $(/bin/pwd)/builddr. But when executing make and sudo make install command, it said no target specified and no makefile found.  

But I was still able to get apptainer --version as 1.3.0 and added the path to bashrc file. The problem was when testing and checking build configuration. I got the errors, ERROR  : Failed to set securebits: Invalid argument
ERROR  : Child exit with status 1. So I am not able to proceed any further. Can anyone guide me a way through this ?
